Description: better fullscreen
 update fullscreen function, to make go fullscreen and back, with F key
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Applied-Upstream: no
Last-Update: 2020-04-28
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/main.cpp
+++ b/main.cpp
@@ -128,6 +128,7 @@
 				{
 				case SDLK_ESCAPE:done = true;break;
 				case SDLK_q: done = true; break;
+				case SDLK_f: win.ToggleFullscreen(); break;
 				}
 			}
 
--- a/win32.cpp
+++ b/win32.cpp
@@ -19,6 +19,13 @@
 {
 }
 
+void Window::ToggleFullscreen() {
+    int FullscreenFlag = SDL_WINDOW_FULLSCREEN_DESKTOP;
+    int isFullscreen = SDL_GetWindowFlags(sdlWindow) & FullscreenFlag;
+    SDL_SetWindowFullscreen(sdlWindow, isFullscreen ? 0 : FullscreenFlag);
+    SDL_ShowCursor(isFullscreen);
+}
+
 bool Window::create(bool bflag)
 {
 	SDL_Init(SDL_INIT_VIDEO);
--- a/win32.hpp
+++ b/win32.hpp
@@ -11,6 +11,7 @@
 SDL_Window *sdlWindow;
 bool create(bool bflag);
 void shut();
+void ToggleFullscreen();
 
 //		HDC getDC();
 
